### PR TITLE
remove openai references from datasets

### PR DESCRIPTION
**Pull request by CloseChoice**
#2389

remove openai from the datasets:
 - gpt4all
 - alpaca
 - code_alpaca

I checked the following datasets additionally but did not found any entries mentioning openai:
    - joke
    - webgpt
    - minimath
    - humaneval_mbpp_codegen_qa
    - humaneval_mbpp_testgen_qa
    - grade_school_math_instructions
    - recipes
    - cmu_wiki_qa
    - oa_wiki_qa_bart_10000row
    - prosocial_dialogue:
    - explain_prosocial

### CODE REVIEW

File: model/model_training/utils.py [hunk 1]
```diff
@@ -364,6 +364,9 @@ def get_dataset(
 
     for data_config in conf.datasets + conf.datasets_extra:
         dataset_name, kwargs = get_dataset_name_and_kwargs_from_data_config(data_config)
+        import pdb
+
+        pdb.set_trace()
```
Comment (sanagno) [inline model/model_training/utils.py:369]: remove this?

Comment (CloseChoice) [reply model/model_training/utils.py]: done

File: model/model_training/custom_datasets/qa_datasets.py [hunk 2]
```diff
@@ -503,6 +507,9 @@ def process_vicuna_conversations(data: list[dict[str, None | str]], input_max_le
         for line in data["conversations"]:
             speaker = line["from"]  # 'human' or 'gpt'
             message = line["value"]
+            # if _filter_by_words(message) is None:
```
Comment (andreaskoepf) [inline model/model_training/custom_datasets/qa_datasets.py:510]: commented, should be removed?

Comment (CloseChoice) [reply model/model_training/custom_datasets/qa_datasets.py]: done

**Review by andreaskoepf — APPROVED** [review]

**Comment by olliestanley** [comment]
I am curious if this has any effect on the oasst dataset(s)?

**Comment by CloseChoice** [comment]
@olliestanley How do you mean that? Oasst datasets are not touched and I did not check them.

**Comment by olliestanley** [comment]
> @olliestanley How do you mean that? Oasst datasets are not touched and I did not check them.

Ok, makes sense! I guess there are probably some references to OpenAI in the oasst data from people copy-pasting ChatGPT responses onto our site, but hopefully only a small amount

**Comment by andreaskoepf** [comment]
@olliestanley I asked Tobias to add this filter since I noticed that a model that was pre-trained (without OA data) replied to some questions with OpenAI statements ... We could do similar filtering for the OA dataset but since we do have full control over the OA data we probably don't need to filter during loading but can directly alter or delete affected messages.

**Pull request merged**